src/components/IstioWizards/IstioWizard.tsx

```diff
@@ -166,18 +171,59 @@ class IstioWizard extends React.Component<Props, State> {
                   host: this.props.serviceName,
                   subset: wkdNameVersion[workload.name]
                 },
-                weight: workload.traffic
+                weight: workload.weight
               };
             })
           }
         ]
+      };
+    }
+
+    if (this.props.type === WIZARD_MATCHING_ROUTING) {
+      // VirtualService from the routes
+      wizardVS.spec = {
+        hosts: [this.props.serviceName],
+        http: this.state.rules.map(rule => {
+          const httpRoute: HTTPRoute = {};
+          const destW: DestinationWeight = {
+            destination: {
+              host: this.props.serviceName
+            }
+          };
+          if (rule.routeType === ROUTE_TYPE.WORKLOAD) {
+            destW.destination.subset = wkdNameVersion[rule.route];
+          }
+          httpRoute.route = [destW];
+          if (rule.matches.length > 0) {
+            httpRoute.match = this.buildHTTPMatchRequest(rule.matches);
+          }
+          return httpRoute;
+        })
+      };
+    }
+
```

Comment:
Could we have more types in the future? The value of hosts are the same and maybe we can change this by a switch case (2º conditional if is not necessary be evaluate if the type is WIZARD_WEIGHTED_ROUTING) What do you think?

Reply:
Will move to a switch case.